enhancements/network/netflow.md

```diff
@@ -0,0 +1,110 @@
+---
+title: OpenShift NetFlow support
+authors:
+  - "@rcarrillocruz"
+reviewers:
+  - "@russellb"
+approvers:
+  - TBD
+creation-date: 2021-01-11
+last-updated: 2020-01-11
+status: provisional
+---
+
+# OpenShift NetFlow support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+NetFlow is a protocol developed by Cisco for collecting IP traffic
+information and monitoring network flow. It is widely used by network departments.
+
+OVS (Open vSwitch) has supported NetFlow for quite some time now.
+OpenShift uses OVN/OVS for OVN-Kubernetes network plugin. 
+OpenShift should have the ability to export cluster traffic as NetFlow data.
+
+## Motivation
+
+* Customers need to be able to get NetFlow data out of OpenShift to be consumed by their collectors.
+* Customers need to be able to see a log of the network traffic that flows thru OpenShift cluster.
+* Customers need to get metrics and monitoring data of the OpenShift cluster network traffic to make decisions on capacity and/or security issues.
+
+### Goals
+
+- Add support for exporting NetFlow data from OpenShift.
+- Add an optional collector that consumes the netflow data and enriches it with OpenShift metadata. 
+- Add Prometheus metrics from NetFlow data collected for visualization and alerting purposes.
+
+### Non-Goals
+
+- This document only applies to OVN-Kubernetes, not OpenShift SDN.
+- This document will not the flow logs store solutions. Typically, customers store the NetFlow data in a data warehouse system (e.g. ClickHouse) or search and analytics system (e.g. ElasticSearch). It is out of the scope of this document to discuss which option should be used and if we should manage that solution. 
+
+## Proposal
+
+The Cluster Network Operator (CNO) should expose options to enable the NetFlow data from OVS bridges.
+OVS supports NetFlow v5, IPFIX and sflow. NetFlow v5 does not support IPv6, therefore we should only consider as protocol options
+both IPFIX/sflow or pick one of them. Sflow is a more performant as it is a sampling protocol and does not rely on templates.
```

Comment:
Up to this point, the enhancement implies NetFlow is the protocol.  Would you like to include a section in this doc about netflow vs. ipfix vs. sflow?  I have some notes on this I can share with you.  I see the goflow README has a high level overview, as well.  I think explaining it in this enhancement would be helpful though.

It seems we may need to support all of them anyway if we want to support customers' own collectors.  Then we can just pick a default to use for our own collector that dumps the info to logs?

... and if we're supporting all the protocols, then maybe the language should be changed throughout the enhancement to say something generic like "network flow collection" instead of "NetFlow" specifically?

Reply:
yeah,makes sense.
sflow has a lesser performance impact at the expense of being sampled (i.e. it won't take all packets).
But I can see customers taking both, so maybe we should change the title to network flow collection as you suggest.

Reply:
done